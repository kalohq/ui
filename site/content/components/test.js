<h1 id="design-tokens">Design Tokens</h1>
<blockquote>
<p><strong>Note</strong>: The files in this directory are autogenerated. See <a href="/config/design-tokens">config/design-tokens</a> for the source.</p>
</blockquote>
