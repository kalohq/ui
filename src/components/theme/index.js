/* @flow */
export {default} from './theme';
