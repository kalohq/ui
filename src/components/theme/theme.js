/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
