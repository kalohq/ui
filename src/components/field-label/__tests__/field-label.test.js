/* @flow */
/* eslint-env jest */
import React from 'react';
import {textMatch} from 'utils/test/enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import FieldLabel from '../field-label';
import Text from '../../text';
import Icon from '../../icon';
import theme from '../../theme';

describe('FieldLabel', () => {
  const defaultProps = {
    htmlFor: 'user-email',
    children: 'Your email',
  };
  const create = props => <FieldLabel {...defaultProps} {...props} />;

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <FieldLabel {...defaultProps} />
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render with a required asterisk', () => {
    const element = create({
      required: true,
    });
    const result = shallow(element);
    const label = result.find(Text);
    expect(textMatch(label, 'Your email *')).toBe(true);
  });

  test('should render with a locked icon', () => {
    const element = create({
      locked: true,
    });
    const result = shallow(element);
    expect(result.find(Icon).length).toBe(1);
  });

  test('should render with a custom icon', () => {
    const element = create({
      icon: 'archive',
    });
    const result = shallow(element);
    expect(result.find(Icon).length).toBe(1);
  });
});
