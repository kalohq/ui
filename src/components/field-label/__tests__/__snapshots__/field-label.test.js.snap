// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldLabel should render shallow component ok 1`] = `
<label
  className="em7k1990 css-bjcoq8"
  color="navy600"
  htmlFor="user-email"
  onClick={undefined}
  size="small"
  style={Object {}}
  title={undefined}
>
  Your email
</label>
`;
